sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/ApproxCountDistinctForIntervals.scala

```diff
@@ -239,4 +221,23 @@ case class ApproxCountDistinctForIntervals(
   override def dataType: DataType = ArrayType(LongType)
 
   override def prettyName: String = "approx_count_distinct_for_intervals"
+
+  override def serialize(obj: Array[Long]): Array[Byte] = {
+    val buffer = ByteBuffer.wrap(new Array(obj.length * Longs.BYTES))
```

Comment:
IIRC `ByteBuffer` is pretty slow for writing, shall we use unsafe writing?

Reply:
Changed to unsafe writing, could you take another look?